### PR TITLE
Kind gitops

### PR DESCRIPTION
**Pull request by NillsF**
Adding kind gitops scenario using helm.
#112

### CODE REVIEW

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+![](../img/local_kind_gitops_helm/05.png)
+
+
+
+# Cluster-level Config vs. Namespace-level Config 
+
+## Cluster-level Config
+
+With Cluster-level GitOps config, the goal is to have "horizontal components" or "management components" deployed on your Kubernetes cluster which will then be used by your applications. Good examples are Service Meshes, Security products, Monitoring solutions, etc. You will not be creating a cluster-level config in this guide. For an example of a cluster-level configuration please refer to the [helm based gitops on AKS scenario](aks_gitops_helm.md).
+
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md:88]:
```suggestion
With Cluster-level GitOps config, the goal is to have "horizontal components" or "management components" deployed on your Kubernetes cluster which will then be used by your applications. Good examples are Service Meshes, Security products, Monitoring solutions, etc. You will not be creating a cluster-level config in this guide. For an example of a cluster-level configuration please refer to either the [Helm-based GitOps on AKS scenario](aks_gitops_helm.md) or the [GKE one](gke_gitops_helm.md).

```

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+## Namespace-level Config
+
+With Namespace-level GitOps config, the goal is to have Kubernetes resources deployed only in the namespace selected. The most obvious use-case here is simply your application and its respective pods, services, ingress routes, etc. In the next section will have the "Hello Arc" application deployed on a dedicated namespace. 
+
+# Azure Arc Kubernetes GitOps Configuration with Helm
+
+## The Mechanism (In a nutshell)
+
+In the process of creating Azure Arc GitOps configuration, [Weaveworks Flux Kubernetes Operator](https://github.com/fluxcd/flux) is deployed on the cluster.
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md:97]:
```suggestion
In the process of creating Azure Arc enabled Kubernetes GitOps configuration, [Weaveworks Flux Kubernetes Operator](https://github.com/fluxcd/flux) is deployed on the cluster. 
```

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+With Namespace-level GitOps config, the goal is to have Kubernetes resources deployed only in the namespace selected. The most obvious use-case here is simply your application and its respective pods, services, ingress routes, etc. In the next section will have the "Hello Arc" application deployed on a dedicated namespace. 
+
+# Azure Arc Kubernetes GitOps Configuration with Helm
+
+## The Mechanism (In a nutshell)
+
+In the process of creating Azure Arc GitOps configuration, [Weaveworks Flux Kubernetes Operator](https://github.com/fluxcd/flux) is deployed on the cluster. 
+
+The Operator is aware to an "HelmRelease" Custom Resource Definition (CRD). This HelmRelease points to a git repo and can optionally contain specific values to input into the helm chart. Due to this configuration, a user can choose to leave the chart values intact or to have different values for different releases.
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md:99]: It is actually the flux-helm operator that is looking at the Helm release so you should point that out.

Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md:99]: More accurately, the HelmRelease is pointing to a Helm chart within a repo

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+    }
+    ```
+    
+    **Note**: It is optional but highly recommended to scope the SP to a specific [Azure subscription and Resource Group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest) 
+# Manually setting up an ingress contoller on kind
+
+The demo application that will be deployed later in this guide relies on an ingress controller. For ingress controllers to work on kind, a specific configuration of the ingress needs to be deployed. For more information related to this, please refer to the [kind documentation](https://kind.sigs.k8s.io/docs/user/ingress/).
+
+## Deployment
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md:67]:
```suggestion
## NGINX Controller Deployment
```

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md (hunk 3 above)
Comment (NillsF) [reply azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md]: adapted

Comment (NillsF) [reply azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md]: I didn't adapt that piece, didn't really know how to reword. I'll take suggestions here, and if we update this statement - we might want to changes this across the different scenarios in the repo.

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+With Namespace-level GitOps config, the goal is to have Kubernetes resources deployed only in the namespace selected. The most obvious use-case here is simply your application and its respective pods, services, ingress routes, etc. In the next section will have the "Hello Arc" application deployed on a dedicated namespace. 
+
+# Azure Arc Kubernetes GitOps Configuration with Helm
+
+## The Mechanism (In a nutshell)
+
+In the process of creating Azure Arc enabled Kubernetes GitOps configuration, [Weaveworks Flux Kubernetes Operator](https://github.com/fluxcd/flux) is deployed on the cluster. 
+
+The Operator is aware to an "HelmRelease" Custom Resource Definition (CRD). This HelmRelease points to a helm chart in a git repo and can optionally contain specific values to input into the helm chart. Due to this configuration, a user can choose to leave the chart values intact or to have different values for different releases.
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md:100]:
```suggestion
The Operator is aware of the "HelmRelease" Custom Resource Definition (CRD). This HelmRelease points to a helm chart in a git repo and can optionally contain specific values to input into the helm chart. Due to this configuration, a user can choose to leave the chart values intact or to have different values for different releases. 
```

File: azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md (hunk 3 above)
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md]:
```suggestion
The Operator is aware of the "HelmRelease" Custom Resource Definition (CRD). This HelmRelease points to a helm chart in a git repo and can optionally contain specific values to input into the helm chart. Due to this configuration, a user can choose to leave the chart values intact or to have different values for different releases. 
```

Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md]:
```suggestion
The Operator is aware of the "HelmRelease" Custom Resource Definition (CRD). This HelmRelease points to a helm chart in a git repo and can optionally contain specific values to input into the helm chart. Due to this configuration, a user can choose to leave the chart values intact or to have different values for different releases.
```

Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_kind_gitops_helm.md]:
```suggestion
The Operator is aware of the "HelmRelease" Custom Resource Definition (CRD). This HelmRelease points to a helm chart in a git repo and can optionally contain specific values to input into the helm chart. Due to this configuration, a user can choose to leave the chart values intact or to have different values for different releases.
```

File: azure_arc_k8s_jumpstart/kind/gitops/helm/az_k8sconfig_helm_aks.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+echo "Create Namespace-level GitOps-Config for deploying the 'Hello Arc' application"
+az k8sconfiguration create \
+--name hello-arc \
+--cluster-name $arcClusterName --resource-group $resourceGroup \
+--operator-instance-name hello-arc --operator-namespace prod \
+--enable-helm-operator --helm-operator-version='0.6.0' \
+--helm-operator-params='--set helm.versions=v3' \
+--repository-url $appClonedRepo \
+--scope namespace --cluster-type connectedClusters \
+--operator-params="--git-poll-interval 3s --git-readonly --git-path=releases/prod"
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/kind/gitops/helm/az_k8sconfig_helm_aks.sh]: change the filename to az_k8sconfig_helm_kind.sh

File: azure_arc_k8s_jumpstart/kind/gitops/helm/az_k8sconfig_helm_cleanup.sh [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+export arcClusterName='<Your Arc cluster name>'
+
+# Deleting GitOps Configurations from Azure Arc Kubernetes cluster
+echo "Deleting GitOps Configurations from Azure Arc Kubernetes cluster"
+az k8sconfiguration delete --name hello-arc --cluster-name $arcClusterName --resource-group $resourceGroup --cluster-type connectedClusters -y
+
+# Cleaning Kubernetes cluster
+echo "Cleaning Kubernetes cluster. You Can safely ignore non-exist resources"
+kubectl delete ns prod
+
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/kind/gitops/helm/az_k8sconfig_helm_cleanup.sh:16]: delete the Arc helm secret as well

File: azure_arc_k8s_jumpstart/kind/gitops/helm/az_k8sconfig_helm_cleanup.sh [hunk 8]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+# <--- Change the following environment variables according to your Azure Service Principle name --->
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/kind/gitops/helm/az_k8sconfig_helm_cleanup.sh:3]: you might want to force az login again using the SPN

**Review by likamrat — CHANGES_REQUESTED** [review]
- Couple of updates on the cleanup process

**Review by likamrat — APPROVED** [review]
LGTM, thx for the contribution!

**Pull request merged**